### ISSUE TITLE
Update docs include syntax for source examples

### ISSUE THREAD

**Issue by tiangolo** [fastapi/fastapi#12554]
### Privileged issue

- [X] I'm @tiangolo or he asked me directly to create an issue here.

### Issue Content

This is a good first contribution. :nerd_face: 

The code examples shown in the docs are actual Python files. They are even tested in CI, that's why you can always copy paste an example and it will always work, the example is tested.

The way those examples are included in the docs used a specific format. But now there's a new format available that is much simpler and easier to use than the previous one, in particular in complex cases, for example when there are examples in multiple versions of Python.

But not all the docs have the new format yet. The docs should use the new format to include examples. That is the task. :nerd_face: 

**It should be done as one PR per page updated.**

## Simple Example

Before, the format was like:

````markdown
```Python hl_lines="3"
{!../../docs_src/first_steps/tutorial001.py!}
```
````

Now the new format looks like:

````markdown
{* ../../docs_src/first_steps/tutorial001.py hl[3] *}
````

* Instead of `{!` and `!}` it uses `{*` and `*}`
* It no longer has a line above with:

````markdown
```Python
````

* And it no longer has a line below with:

````markdown
```
````

* The highlight is no longer a line with e.g. `hl_lines="3"` (to highlight line 3), but instead in the same line there's a `hl[3]`.

An example PR: https://github.com/fastapi/fastapi/pull/12552

## Multiple Python Versions

There are some cases where there are variants of the same example for multiple versions of Python, or for using `Annotated` or not.

In those cases, the current include examples have syntax for tabs, and notes saying `Annotated` should be preferred. For example:

````markdown
//// tab | Python 3.9+

```Python hl_lines="4  8  12"
{!> ../../docs_src/security/tutorial006_an_py39.py!}
```

////

//// tab | Python 3.8+

```Python hl_lines="2  7  11"
{!> ../../docs_src/security/tutorial006_an.py!}
```

////

//// tab | Python 3.8+ non-Annotated

/// tip

Prefer to use the `Annotated` version if possible.

///

```Python hl_lines="2  6  10"
{!> ../../docs_src/security/tutorial006.py!}
```

////
````

In these cases, it should be updated to only include the first one (the others will be included automatically :sunglasses: ):

````markdown
{* ../../docs_src/security/tutorial006_an_py39.py hl[4,8,12] *}
````

* The syntax for tabs is also removed, all the other variants are included automatically.
* The highlight lines are included for that same first file, the fragment with `hl_lines="4  8  12"` is replaced with `hl[4,8,12]`

An example PR: https://github.com/fastapi/fastapi/pull/12553

### Simple Lines

When there's a fragment like:

````markdown
hl_lines="4  8  12"
````

That means it is highlighting the lines 4, 8, and 12.

The new syntax is on the same include line:

````markdown
hl[4,8,12]
````

* It separates individual lines by commas.
* It uses `hl`, with square brackets around.

### Line Ranges

When there are line ranges, like:

````markdown
hl_lines="4-6"
````

That means it is highlighting lines from 4 to 6 (so, 4, 5, and 6).

The new syntax uses `:` instead of `-` for the ranges:

````markdown
hl[4:6]
````

### Multiple Highlights

There are some highlights that include individual lines and also line ranges, for example the old syntax was:

````markdown
hl_lines="2  4-6  8-11  13"
````

That means it is highlighting:

* Line 2
* Lines from 4 to 6 (so, 4, 5, and 6)
* Lines from 8 to 11 (so, 8, 9, 10, and 11)
* Line 13

The new syntax separates by commas instead of spaces:

````markdown
hl[2,4:6,8:11,13]
````

## Include Specific Lines

In some cases, there are specific lines included instead of the entire file.

For example, the old syntax was:

````markdown
```Python hl_lines="7"
{!> ../../docs_src/separate_openapi_schemas/tutorial001_py310.py[ln:1-7]!}

# Code below omitted 👇
```
````

In this example, the lines included are from line 1 to line 7 (lines 1, 2, 3, 4, 5, 6, 7). In the old syntax, it's defined with the fragment:

````markdown
[ln:1-7]
````

In the new syntax, the included code from above would be:

````markdown
{* ../../docs_src/separate_openapi_schemas/tutorial001_py310.py ln[1:7] hl[7] *}
````

* The lines to include that were defined with the fragment `[ln:1-7]`, are now defined with `ln[1:7]`

The new syntax `ln` as in `ln[1:7]` also supports multiple lines and ranges to include.

### Comments Between Line Ranges

In the old syntax, when there are ranges of code included, there are comments like:

````markdown
# Code below omitted 👇
````

The new syntax generates those comments automatically based on the line ranges.

### Real Example

A more real example of the include with the old syntax looked like this:

````markdown
//// tab | Python 3.10+

```Python hl_lines="7"
{!> ../../docs_src/separate_openapi_schemas/tutorial001_py310.py[ln:1-7]!}

# Code below omitted 👇
```

<details>
<summary>👀 Full file preview</summary>

```Python
{!> ../../docs_src/separate_openapi_schemas/tutorial001_py310.py!}
```

</details>

////

//// tab | Python 3.9+

```Python hl_lines="9"
{!> ../../docs_src/separate_openapi_schemas/tutorial001_py39.py[ln:1-9]!}

# Code below omitted 👇
```

<details>
<summary>👀 Full file preview</summary>

```Python
{!> ../../docs_src/separate_openapi_schemas/tutorial001_py39.py!}
```

</details>

////

//// tab | Python 3.8+

```Python hl_lines="9"
{!> ../../docs_src/separate_openapi_schemas/tutorial001.py[ln:1-9]!}

# Code below omitted 👇
```

<details>
<summary>👀 Full file preview</summary>

```Python
{!> ../../docs_src/separate_openapi_schemas/tutorial001.py!}
```

</details>

////
````

In the new syntax, that is replaced with this:

````markdown
{* ../../docs_src/separate_openapi_schemas/tutorial001_py310.py ln[1:7] hl[7] *}
````

* The only file that needs to be included and defined is the first one, and the lines to include and highlight are also needed for the first file.
* All the other file includes, full file preview, comments, etc. are generated automatically.

---

An example PR: https://github.com/fastapi/fastapi/pull/12555

## Help

Do you want to help? Please do!

Remember **it should be done as one PR per page updated.**

If you see a page that doesn't fit these cases, leave it as is, I'll take care of it later.

Before submitting a PR, check if there's another one already handling that file.

Please name the PR including the file path, for example:

````markdown
📝 Update includes for `docs/tutorial/create-db-and-table.md`
````

**Comment by raj-298585** [comment]
Osis

**Comment by raj-298585** [comment]
Ieje

**Comment by FarhanAliRaza** [comment]
i want to contribute 
Will the issue be assigned?

**Comment by tiangolo** [comment]
@FarhanAliRaza the issue is not assigned, there are many PRs that would come out of this, one PR per page fixed, so, you can just take a page that has the old syntax, check there are no other PRs handling it, then update the syntax, and make a PR for it.

**Comment by cookabc** [comment]
> @FarhanAliRaza the issue is not assigned, there are many PRs that would come out of this, one PR per page fixed, so, you can just take a page that has the old syntax, check there are no other PRs handling it, then update the syntax, and make a PR for it.

the game is kicked off 😄

**Comment by FarhanAliRaza** [comment]
it kind of looks it can be automated using script

**Comment by FarhanAliRaza** [comment]
what about cases like this that has title ```Python hl_lines="1  3" title="app/routers/users.py"
{!../../docs_src/bigger_applications/app/routers/users.py!}
```

**Comment by philipokiokio** [comment]
@FarhanAliRaza  do you know why your CI/CD passes successfully two of mine have failed now.

**Comment by FarhanAliRaza** [comment]
@philipokiokio, your commits look fine for me . This is the error https://github.com/fastapi/fastapi/actions/runs/11533792038/job/32107125447?pr=12560

**Comment by philipokiokio** [comment]
@FarhanAliRaza cool, thanks man.

**Comment by FarhanAliRaza** [comment]
> what about cases like this that has title `Python hl_lines="1 3" title="app/routers/users.py" {!../../docs_src/bigger_applications/app/routers/users.py!}`

@tiangolo

**Comment by krishnamadhavan** [comment]
@FarhanAliRaza As suggested by @tiangolo, it'd be easier for everyone to understand which pages you're working on, if you can set the title of your PR with the file path.

Thanks for understanding.

A reference PR from @tiangolo for your perusal: https://github.com/fastapi/fastapi/pull/12552

**Comment by philipokiokio** [comment]
Hi @krishnamadhavan 

so when tabs were used above some code was deleted. is this the expected contribution. example

`#### Tuple and Set

You would do the same to declare `tuple`s and `set`s:

//// tab | Python 3.9+

{* ../../docs_src/python_types/tutorial007_py39.py hl[1] *}


////

//// tab | Python 3.8+


{* ../../docs_src/python_types/tutorial007.py hl[1,4] *}

////`

unlike the example above there was a tutorial for py 3.10

**Comment by FarhanAliRaza** [comment]
> @FarhanAliRaza As suggested by @tiangolo, it'd be easier for everyone to understand which pages you're working on, if you can set the title of your PR with the file path.
> 
> Thanks for understanding.
> 
> A reference PR from @tiangolo for your perusal: https://github.com/fastapi/fastapi/pull/12552

Updated.
Can you answer my above question?

**Comment by lucaromagnoli** [comment]
Hope I got the tabs/ multi python version right :)

**Comment by krishnamadhavan** [comment]
> Hi @krishnamadhavan
> 
> so when tabs were used above some code was deleted. is this the expected contribution. example
> 
> ```
> 
> You would do the same to declare `tuple`s and `set`s:
> 
> //// tab | Python 3.9+
> 
> {* ../../docs_src/python_types/tutorial007_py39.py hl[1] *}
> 
> 
> ////
> 
> //// tab | Python 3.8+
> 
> 
> {* ../../docs_src/python_types/tutorial007.py hl[1,4] *}
> 
> ////```
> 
> 
> 
> 
> unlike the example above there was a tutorial for py 3.10
> ```

@philipokiokio I believe you're talking about [this PR](https://github.com/fastapi/fastapi/pull/12555/files) from @tiangolo . If you're mentioning about the `<details>` and `<summary>` block, it was used to collapse and expand.

But since we're combining the different version definitions into a single line, I think we should be good there.

**Comment by lucaromagnoli** [comment]
Hi @krishnamadhavan 

I just added  PR #12566. Would you be able to check if I got the multiple python versions / tabs bit right?

**Comment by krishnamadhavan** [comment]
> I just added PR #12566. Is anyone able to check if I got the multiple python versions / tabs bit right so I can crack on with more PRs?

I'm also a new contributor. But, yes, it looks good to me :)

**Comment by lucaromagnoli** [comment]
> > I just added PR #12566. Is anyone able to check if I got the multiple python versions / tabs bit right so I can crack on with more PRs?
> 
> I'm also a new contributor. But, yes, it looks good to me :)

Sorry for some reason I thought you were part of the main contributors :)

**Comment by lucaromagnoli** [comment]
@philipokiokio Don't mean to be a pain but would you be able to include the path to the files you changed in the PR title using the same format as the example PR #12555 so it's easier to figure out which files have been changed please? :)

**Comment by krishnamadhavan** [comment]
> @FarhanAliRaza do you know why your CI/CD passes successfully two of mine have failed now.

@philipokiokio This is the root cause of the issue on one of your PRs.

https://github.com/fastapi/fastapi/actions/runs/11533755181/job/32107030159#step:10:45

**Comment by abejaranoh** [comment]
I just created this one #12585 ... i hope is fine :)

**Comment by tiangolo** [comment]
Thanks everyone for the help here! :bow: :rocket: 

If you see a page that doesn't fit the cases described above, leave it as is, I'll take care of it later. It's probably that I don't have an established way to handle that case yet and I'll have to update it later. :nerd_face:

**Comment by kantandane** [comment]
> In these cases, it should be updated to only include the first one (the others will be included automatically 😎 ):
In my PRs preview pages, I indeed get automatically the variants but we lose the highlights on the variants code, is it intentional ? did I miss something ?

**Comment by antonyare93** [comment]
Can anyone help me find out why the tests are failing
https://github.com/fastapi/fastapi/actions/runs/11536985036/job/32113864270?pr=12602

**Comment by krishnamadhavan** [comment]
> Can anyone help me find out why the tests are failing https://github.com/fastapi/fastapi/actions/runs/11536985036/job/32113864270?pr=12602

This is what GitHub CI says.

https://github.com/fastapi/fastapi/actions/runs/11536985036/job/32113864270?pr=12602#step:10:45


But when I took a look at your PR, I can see a `>` character. I've commented on your PR hoping that'd help.

**Comment by lucaromagnoli** [comment]
It seems there is an issue in the build caused by 

`PendingDeprecationWarning: Please use import python_multipart instead`

I would hold off until it gets fixed by the main contributors :)

**Comment by manvillej** [comment]
Hey @tiangolo, are we just waiting on #12627 to start merging the new syntax PRs?

**Comment by tiangolo** [comment]
@manvillej no worries, we are already handling the new PRs (by "we" I mean @alejsdev :sweat_smile: :nerd_face: ).

And that PR was just merged so no more false positive errors in CI.

**Comment by fegmorte** [comment]
Hi @tiangolo, 
For a good starting point, I would like to contribute to update fastapi docs. I am french so I have been starting to update the french doc and in the order of the directory. First file `fr/docs/advanced/additional-responses.md` I want to update, contains Python directives : `Python hl_lines="3"` but also Json directives like that : `JSON hl_lines="4-16"`
Is there something to do with Json directives also ? Or I don't mind about Json comment.

**Comment by Joao-Pedro-P-Holanda** [comment]
@tiangolo I don't know if this issue is the correct place, but a question Discussion also doesn't seem as a best fit. If its relevant i can later move to a separate issue.

One more thing that was updated in Material for MkDocs was that admonitions with some text defined with a pipe `|` don't need double quotes anymore, but currently most of the docs still have it.

So the code:

```md
/// info| "Nota"
...
///
```` 

will render with the double quotes.
![image](https://github.com/user-attachments/assets/9b7c9c38-44d5-4a58-a58a-d85a1b578673)

Whereas:

```md
/// info | Nota
...
///
```
![image](https://github.com/user-attachments/assets/d1b91c71-3124-434f-8501-4739f8a9738f)

renders normally,

I used a Vim regular expression to find all the occurences of this pattern, and if it is valid problem, i want to contribute replacing by texts without double quotes.

```vim
" finds double quoted texts after "/// | " then replaces by the text inside the quotes 
:%s/\/\/\/ .* | \zs"\(.*\)"\ze/\1
```
 
This regular expression can also be applied to every file in a folder using Vim with the command `args` : 
```sh
# opening all docs with vim
vim docs/<language>/docs
```
```vim
" adding all files of a folder to the arguments
:args <folder_name>/*
" updating with the regular expression
:argdo %s/\/\/\/ .* | \zs"\(.*\)"\ze/\1/g | update
```` 

Some example of this behavior is in the portuguese translation of `body-multiple-params`, [here](https://fastapi.tiangolo.com/pt/tutorial/body-multiple-params/#misture-path-query-e-parametros-de-corpo).

**Comment by SalimNhiri** [comment]
Update includes for docs/tutorial/body-multiple-params.md #12717

**Comment by SalimNhiri** [comment]
#12717 Update includes for docs/tutorial/body-multiple-params.md #12717

**Comment by tiangolo** [comment]
This is finished now, thanks everyone for the help! :cake: :coffee:
